mode/clike/clike.js

```diff
@@ -519,6 +519,16 @@ CodeMirror.defineMode("clike", function(config, parserConfig) {
         stream.eatWhile(/[\w\$_]/);
         return "meta";
       },
+      "{": function(stream, state) {
+        // match a single full line with params of multi-line anonymous fn, e.g.:
+        // { a =>\n
+        // { (aa: Int, bb: Int) =>\n
+        var singleLineParams = /^(\w|[:.,() ])+=>$/;
+        if (!stream.match(singleLineParams, true)) // eat the line, if matches
```

Comment:
> but then, how to conveniently request tokening of the rest of the current line?

You don't need to, I think. You just change your hook to run on the `"="` character instead, and when it sees a `">"` ahead, it changes the current context to not try and align its body.